core/src/main/java/bisq/core/dao/node/full/RpcService.java

```diff
@@ -246,8 +254,10 @@ private static RawTx getTxFromRawTransaction(RawTransaction rawDtoTx,
                     // To maintain backwards compatibility when serializing and hashing the DAO state,
                     // segwit pubKeys are only extracted for the first input, as this will always be a
                     // BSQ input. Later inputs might be segwit BTC, which would have had a null pubKey
-                    // recorded in the DAO state prior to the segwit upgrade of the RPC client.
-                    String pubKeyAsHex = extractPubKeyAsHex(rawInput, rawInput == rawDtoTx.getVIn().get(0));
+                    // recorded in the DAO state prior to the segwit upgrade of the RPC client. Spurious
+                    // segwit BSQ inputs in txs mined prior to the upgrade also require exclusion.
+                    String pubKeyAsHex = extractPubKeyAsHex(rawInput, rawInput == rawDtoTx.getVIn().get(0) &&
+                            !BSQ_TXS_DISALLOWING_SEGWIT_PUB_KEYS.contains(txId));
```

Comment:
Wouldn't it be better to use a block height at which we activate the support for segwit inputs. It is technically a hard fork which would become activated at that block height.
That should also cover the past tx(s) with a segwit input.

Reply:
Yes, perhaps that would be better. I'll push a change to do that shortly, as well as no longer excluding all but the first segwit input pubkey (to be on the safe side and consistent with the original pubkey extraction for P2PKH inputs).